lib/Dancer2/Core/Role/SessionFactory.pm

```diff
@@ -163,7 +164,7 @@ sub retrieve {
     my %args = ( id => $id, );
 
     $args{data} = $data
-      if $data and ref $data eq 'HASH';
+      if $data and ref $data and reftype $data eq 'HASH';
```

Comment:
I should look how/if the session data is serialized. Passing objects instead of pure hashes could make things break.